src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -56,5 +73,8 @@ export class DeviceSourceManager {
         this.getDeviceSource = this._deviceSourceManager.getDeviceSource;
         this.getDeviceSources = this._deviceSourceManager.getDeviceSources;
         this.getDevices = this._deviceSourceManager.getDevices;
+        this.enableEvents = this._deviceSourceManager.enableEvents;
+        this.disableEvents = this._deviceSourceManager.disableEvents;
+        this.dispose = this._deviceSourceManager.dispose;
```

Comment:
Does this work? `InternalDeviceSourceManager` is basically a singleton right now. If dispose is called, it will dispose the singleton, which isn't what you want? I think you may want to use a ref counting scheme or do something else entirely.

Reply:
I see what you mean.  Lemme implement a ref count to avoid it getting disposed of before it needs to.

Reply:
I ended up writing dispose to just dispose of the DSM instance and left the InternalDSM to handle the proper disposal when the engine calls dispose

Reply:
The dispose on the DeviceSourceManager will now just remove any observers and remove any references to parts of the InternalDeviceSourceManager, preventing any changes to other instances of the DSM.